### PR TITLE
Release jul19

### PR DESCRIPTION
**Pull request by teresanunez**
This is a PR to master for the official [Jul19 release](https://github.com/sardana-org/sardana/milestone/6). Final tests (automatic and manual) should be done based on it before accepting the merge.

So, this is a call for help from @sardana-org/integrators and any other person willing to help to test this branch ASAP in their respective systems.

For the moment the following systems are going to be tested thoroughly (feel free to add more to the list if you are willing to do the tests):

- [x] openSuse 11.1 (#1156)
- [x] Debian 9 (#1155)
- [x] CentOS 7 (#1154)
- [x] Windows 10 (#1157)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,91 @@ All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 This file follows the formats and conventions from [keepachangelog.com]
 
-## [Unreleased]
+## [2.8.0] 2019-06-18
```
Comment (reszelaz) [inline CHANGELOG.md]: This needs to be a link to a diff between two tags - see the bottom of the CHANGELOG.md file.

**Comment by amilan** [comment]
Hi all, 

I have run the tests listed in here:
https://github.com/sardana-org/sardana/blob/develop/doc/how_to_release.md

In CentOS7 using Taurus 4.5.1 and so far I have only found a couple of small issues that seems to be related with my setup. I will come back with more info once I will fix them.

Cheers,
Antonio.

**Comment by reszelaz** [comment]
Thanks for the tests @amilan!
In the last two releases we used to create a dedicated issue for each of the platform tests, e.g. 
#1057 or #903. And then reference them with with a checkbox list in the release PR message: https://github.com/sardana-org/sardana/pull/904#issue-207015150.

I personally like this approach cause then we can add more details in the test issue itself at the same time not adding to much noise the the release PR. If you agree, we could change the point 6 of the [how to release manual](https://github.com/sardana-org/sardana/blob/develop/doc/how_to_release.md):

> Perform manual tests (see checklist below). You may use the CI artifacts (e.g., from appveyor) and post the results in the comments of the PR.

**Comment by teresanunez** [comment]
Hi @reszelaz, @amilan is not working today (holidays in Sweden). IMO it is fine to create the dedicated 
issues for the checks and also to change the point 6 of the 'how to release' manual.

**Comment by reszelaz** [comment]
Great!

> and also to change the point 6 of the 'how to release' manual.

Will you also take care of adapting the how-to. Go ahead and push the changes direclty to the repo.

Also related to that, I think we could move this document to the Wiki. What do you think?

**Comment by teresanunez** [comment]
> > and also to change the point 6 of the 'how to release' manual.
> 
> Will you also take care of adapting the how-to. Go ahead and push the changes direclty to the repo.
> 

Thanks, just done.

> Also related to that, I think we could move this document to the Wiki. What do you think?

I think it is a good idea.

**Comment by reszelaz** [comment]
Great! I see that the release is advancing!

I was reviewing the pending PR in the milestone. There were two of them:
* Use PY_OBJECT for signal signatures: https://github.com/sardana-org/sardana/pull/1082
* Implement MGConfiguration new API: https://github.com/sardana-org/sardana/pull/867

Both of them however still require work to be done (the first one a backport to taurus 3 branch and the second one an agreement on the API and small changes). In order to not wait with the relaese for them I propose to remove them from the milestone. The idea is to work on them simultaneously to the release works, and integrate into develop ASAP. We aim to integrate them before we start migrating to Python 3 so these features will be available in support-2.x branch.

Regarding the manual tests, from Alba we will do the openSUSE (@rhomspuron) and Windows (@reszelaz). We will start next week. Anyone volunteers for Debian 9? :)

**Comment by teresanunez** [comment]
hI @reszelaz, I have already started the tests in debian9. My results are in the issue created for that.
I did not make the last tests (related with taurus)  due to some conflicts related to my computer.

**Comment by amilan** [comment]
Hi all, 
I'm back from a very short holidays.
I also like the approach commented by @reszelaz and I see that @teresanunez already added the issues. I will try to continue with the test today (and comment it on its PR :P).

Cheers.

**Comment by teresanunez** [comment]
I have finished my tests in debian9. Results are in the issue.

**Comment by amilan** [comment]
Tests done for CentOS. Only missing the tests for sardanaeditor.
Results in #1154.

**Comment by reszelaz** [comment]
I'm about to start tests on Windows. Just to confirm, we all do tests using the latest taurus develop branch, right? I assume that based on the https://github.com/taurus-org/taurus/pull/931 which reported @teresanunez (I think that this bug was not there in Taurus 4.5).

**Comment by teresanunez** [comment]
Hi Zibi, yes, I did the tests with the version in develop from the date I started them (Friday last
 week).

**Comment by reszelaz** [comment]
>> Also related to that, I think we could move this document to the Wiki. What do you think?

> I think it is a good idea.

I have moved the how to to the Wiki:
https://github.com/sardana-org/sardana/wiki/Release-Sardana
https://github.com/sardana-org/sardana/wiki/Do-manual-tests

Feel free to edit it directly with the new experience from this release. Thanks!

**Comment by reszelaz** [comment]
I have just finished the tests on Windows. There was just one regression found (output block does not work on spock) and I have already proposed a fix: #1160.
Also one bug seems to be fixed (interactive macros now works): #347 
I think this results are satisfactory so I mark the Windows tests as checked in the PR description.

**Comment by reszelaz** [comment]
Also tests on openSuste got OK for the release.

Now just waiting for green light from Debian 9 and CentOS? @teresanunez, @amilan and @daneos, what do you think?

**Comment by teresanunez** [comment]
> Also tests on openSuste got OK for the release.
> 
> Now just waiting for green light from Debian 9 and CentOS? @teresanunez, @amilan and @daneos, what do you think?

Hi @reszelaz, I have just closed the debian9 issue #1155.

**Comment by amilan** [comment]
> Now just waiting for green light from Debian 9 and CentOS? @teresanunez, @amilan and @daneos, what do you think?

Hi, I just commented in the CentOS issue that for me the tests are OK except the one for the showscan, but I don't consider that as a showstopper. If you agree we can open a new issue for it, mark the tests as done and continue with the release.

**Comment by teresanunez** [comment]
> > Now just waiting for green light from Debian 9 and CentOS? @teresanunez, @amilan and @daneos, what do you think?
> 
> Hi, I just commented in the CentOS issue that for me the tests are OK except the one for the showscan, but I don't consider that as a showstopper. If you agree we can open a new issue for it, mark the tests as done and continue with the release.

Fine for me, thanks.

**Comment by amilan** [comment]
Great, I have market the CentOS tests as done.

**Pull request merged**